`azurerm_databricks_access_connector`: add support for UAI (#21059)
hashicorp/terraform-provider-azurerm: website/docs/r/databricks_access_connector.html.markdown

An `identity` block supports the following:

> * `type` - (Required) Specifies the type of Managed Service Identity that should be configured on this Access Connector. Possible values are `SystemAssigned`, `UserAssigned`, `SystemAssigned, UserAssigned` (to enable both).
> 
> * `identity_ids` - (Optional) Specifies a list of User Assigned Managed Identity IDs to be assigned to this Access Connector.
> 
> ~> **NOTE:** This is required when `type` is set to `UserAssigned` or `SystemAssigned, UserAssigned`.

Reviewer: at this time the Service doesn't actually support `SystemAssigned, UserAssigned` even though the Swagger defines it (I've sent https://github.com/Azure/azure-rest-api-specs/pull/23370 to fix that) - so we can make this:

```suggestion
* `type` - (Required) Specifies the type of Managed Service Identity that should be configured on this Access Connector. Possible values are `SystemAssigned` and `UserAssigned`.

* `identity_ids` - (Optional) Specifies a list of User Assigned Managed Identity IDs to be assigned to this Access Connector.

~> **NOTE:** This is required when `type` is set to `UserAssigned`.
```

Reviewer 2: Fixed.

Author: @WodansSon haha. Thanks man. In a rush to merge this one? :) I'd get to it tomorrow otherwise.

Reviewer 2: Yes this is an APEX ask resource so there is release by date assigned to it... I would have normally asked, but time is of the essence at the moment... sorry 🙂
